### DISCUSSION TITLE
Handling `go` software differently

### DISCUSSION THREAD

**Discussion by v1gnesh** [zopencommunity/meta#625] [Ideas]
By default, everywhere else, `go install org/repo@latest` produces just one binary, and is placed in `$HOME/go/bin`.
I think we can cut the zopen install steps for go projects (identified as having `ZOPEN_COMP=GO` in buildenv), and pile the binaries in the preferred `.../bin` (`/usr/local/bin`, whatever) directory, and not pay the administrative/setup cost that other C/C++ projects have to pay (as they come with man/lib/etc.).

**Comment by DevonianTeuchter** [comment]
Or just have 'go install' behave as it does everywhere else and install to $home/go...? 
Tbh, not entirely sure why you would need/want to install into the zopen environment if you're not using the zopen tooling - zopen can't manage/understand it (think how SMPE behaves if you modify files it's trying to control, it doesn't much like it 😁). 
Having never dabbled with go, does go have its own repos or where does the "thing" that "go install" actually installs originate?  Would having it in zopenrootfs/opt/go/bin be an idea...?

**Reply by v1gnesh** [reply]
> Or just have 'go install' behave as it does everywhere else and install to $home/go...?

Ideally, yes. However, `zopen build` is useful for doing stuff (patching dependencies, etc.) in the configure stage.
Try cloning buildkiteport and building it with zopen build.
It's an example where a go workspace is used to pull in and patch dependencies.
What it also seems to be doing is, because of the workspace, it caches stuff in `buildkiteport/go/...` instead of `$HOME/go/...`.

By doing just `go build` or `go install` for projects where zOS-specific patches are required, it keeps all of it in `$HOME/go`.
So things can get messy real fast.
As a general read, I found [this ](https://boyter.org/posts/how-to-start-go-project-2023/) helpful.

**Reply by v1gnesh** [reply]
> Where does it originate?

One of the `pkg` `mod` or `cache` directories in `$HOME/go`.

> Would having it in zopenrootfs/opt/go/bin be an idea...?

Yup, any central location to lump all go binaries will do.
